### PR TITLE
React homework №1. Application structure.

### PR DESCRIPTION
**Pull request by kkulahin**
# React Task 1

### Task:

- Build simple layout using components with React, which are: **header, content (main section) and footer**.

- Create store variable and pass it down to components using props.

### Requirements

- use **create-react-app** as a boilerplate for your application (this one is optional, you can create your own Webpack config)

- in your project please follow suggested folder structure: (here you can also find your own way on how to make a decent folder structure, this is just my proposal)

img

**index.js** file should just import the ```<App />``` component and simply call the ```ReactDOM.render method```. Also you need to store a variable here called **store**. This variable should be an object with required fields: **user** (contains **firstName** and **lastName**) and **avatar** (contains **url** and **alt**).

**index.css**(index.module.scss/index.scss) should just contain all the **basic styles** for your app (like rules for html, body)
img_1

Each React component should have its own folder with the jsx and relevant stylesheets

for example

img_2

Then you need to find a way on how to pass the store value through an App component to use this store value in a child components (Header, Content). Footer Component can be hardcoded for now.

Here is the diagram which explains basic data flow

img_3

### Optional tasks

- use SCSS as a preprocessor

- use CSS modules

- create an additional index.js file for each component which will just re-export your component. This is always done to make the import shorter (no hints here, try to research it by yourself)

**You can use yarn or npm. It's your choice**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    "stylelint": "npx stylelint src/**/*.s{c,a}ss",
+    "lint:fix": "tsc --noEmit && eslint --fix src/**/*.{js,ts,tsx} --no-error-on-unmatched-pattern",
+    "stylelint:fix": "npx stylelint --fix src/**/*.s{c,a}ss",
+    "lint:all": "npm run lint:fix && npm run stylelint:fix",
+    "start:dev": "webpack serve --config=configs/webpack/dev.js --env NODE_ENV=development --env BUILD_TYPE=local",
+    "start:prod": "npm run build && npm start",
+    "start": "node express.js"
+  },
+  "devDependencies": {
+    "@babel/cli": "^7.13.14",
```
Comment (MishkanJigan) [inline package.json]: It is better not to make dependencies dynamic.

File: src/components/Footer/Footer.tsx [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import * as React from "react";
+import "./footer.module.scss";
+
+const Footer: React.FC = () => {
```
Comment (MishkanJigan) [inline src/components/Footer/Footer.tsx]: In this case we can remove React.FC, because there are no props here.

File: package.json (hunk 1 above)
Comment (kkulahin) [reply package.json]: Fixed

File: src/components/Footer/Footer.tsx (hunk 2 above)
Comment (kkulahin) [reply src/components/Footer/Footer.tsx]: Fixed

**Pull request merged**